### PR TITLE
Set start and goal as float

### PR DESCRIPTION
**Pull request by BriceRenaudeau**
* Create a `worldToMapContinous` to compute the decimals for the world to map conversion.
* Change the `setGoal` to use floats as inputs and `static_cast` to unsigned int's for the `getIndex` method.
* Do the same with setStart as well

---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #3955 and #4249 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | sim |
| Does this PR contain AI generated software? | No |

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Comment by BriceRenaudeau** [comment]
CI failure not related.

File: nav2_costmap_2d/src/costmap_2d.cpp [hunk 1]
```diff
@@ -296,6 +296,21 @@ bool Costmap2D::worldToMap(double wx, double wy, unsigned int & mx, unsigned int
   return false;
 }
 
+bool Costmap2D::worldToMapContinuous(double wx, double wy, float & mx, float & my) const
+{
+  if (wx < origin_x_ || wy < origin_y_) {
+    return false;
+  }
+
+  mx = (wx - origin_x_) / resolution_ + 0.5;
```
Comment (SteveMacenski) [inline nav2_costmap_2d/src/costmap_2d.cpp:305]: Cast the origin values to float before running this operation. I've been burned alot recently in MPPI/Smac with mixing types around and getting unexpected outcomes or massive performance hits

Comment (SteveMacenski) [reply nav2_costmap_2d/src/costmap_2d.cpp:305]: `+ 0.5f`

**Comment by SteveMacenski** [comment]
@BriceRenaudeau the first 2 failures seem related to this PR. I've never before seen those tests fail and they're targeting Smac

**Comment by BriceRenaudeau** [comment]
> @BriceRenaudeau the first 2 failures seem related to this PR. I've never before seen those tests fail and they're targeting Smac

Le planner returns: "Start occupied" on the verySmallPath tests.

File: nav2_costmap_2d/src/costmap_2d.cpp (hunk 1 above)
Comment (BriceRenaudeau) [reply nav2_costmap_2d/src/costmap_2d.cpp]: Wz, origin_x and resolution are double, I made the static cast on the result.

**Comment by SteveMacenski** [comment]
Wonderful!

**Pull request merged**